### DISCUSSION TITLE
Code slicing over the AST

### DISCUSSION THREAD

**Discussion by antonpibm** [tree-sitter/tree-sitter#2832] [Q&A] [answered]
Hi everyone

Is there a code slicing capability over the AST? this means given a seed node with an identifier of interest (e.g. function or variable), generate a subtree with all nodes affected by this identifier
This means that if the seed identifier is a variable - expand backwards and forward. 
backwards - on its declaration, then on the variables used for the declaration etc. 
forward - on function calls involving it, then on the return values, then on their function calls etc

**Comment by ObserverOfTime** [comment, marked as answer]
No. Tree-sitter can't do semantic analysis.
